### ISSUE TITLE
Implement Inference Mode for Fast Backtesting

### ISSUE THREAD

**Issue by alexoms** [OptimalMatch/temporal-trading-agents#85]
## Overview
Implemented inference mode to enable fast backtesting using cached models without training, reducing prediction time from minutes to 7-10 seconds.

### 1. Inference Mode Implementation
- **Feature**: Use cached PyTorch models for predictions without training or fine-tuning
- **Performance**: 7-10 second predictions vs several minutes with training
- **Use Case**: Quick parameter testing when speed matters more than having the latest data

### 2. Frontend Changes
**Files Modified**: `frontend/src/pages/BacktestPage.jsx`, `frontend/src/pages/AnalyzePage.jsx`
- Added inference mode checkbox to backtest configuration form
- Added clear UI description explaining tradeoffs
- Connected checkbox state to API requests

### 3. Backend Changes
**Files Modified**: `backend/models.py`, `backend/main.py`, `backend/backtesting_engine.py`
- Added `inference_mode: bool` field to `BacktestConfig` Pydantic model
- Implemented inference path in backtest engine that loads cached models
- Multi-Timeframe strategy properly skipped in inference mode (would require training)

### 4. Ensemble Prediction Updates
**Files Modified**: `examples/crypto_ensemble_forecast.py`, `strategies/strategy_utils.py`
- Added `cutoff_date` parameter to `make_ensemble_predictions()` function
- Added `cutoff_date` parameter to `inference_ensemble()` function
- Backtest engine passes first date of backtest period as cutoff
- Data filtered to cutoff before feature extraction

### Bug 1: Forward-Looking Bias in Backtests
**Severity**: Critical - invalidated backtest results
**Root Cause**: 
- Line 467 in `crypto_ensemble_forecast.py` fetched data up to TODAY
- During July 2025 backtest, models saw November 2025 prices ($106k)
- Predicted drops to $84k-$94k (-20% to -23%), generating only SELL signals (0 buy, 63 sell)

**Fix**:
- Added `cutoff_date` parameter throughout prediction pipeline
- Filter data after fetching: `df_latest = df_latest[df_latest.index <= cutoff_date]`
- Models now only see data available at the simulated time period

**Impact**: Ensures temporal integrity of all backtests

### Bug 2: Mean Reversion Column Name Mismatch
**Severity**: High - strategy completely failed during backtests
**Error**: `WARNING: Mean Reversion strategy failed: 'Close'` (KeyError)
**Root Cause**: 
- Strategy code accessed `df['Close']` (uppercase)
- Backtest engine provided `df['close']` (lowercase)

**Fix**:
- Added dynamic column detection: `close_col = 'Close' if 'Close' in df.columns else 'close'`
- Applied to all data access points in `strategies/mean_reversion_strategy.py`

**Impact**: Mean Reversion strategy now works in both standalone and backtest modes

### Data Flow with Temporal Cutoff
```
BacktestEngine (line 472)
  ↓ cutoff_date = pd.to_datetime(price_data['date'].iloc[0])
inference_ensemble() (line 243)
  ↓ passes cutoff_date
make_ensemble_predictions() (line 482-485)
  ↓ df_latest = df_latest[df_latest.index <= cutoff_date]
Feature Extraction (only historical data)
  ↓
Model Predictions (temporally valid)
```

### Files Changed
1. `backend/models.py` - Added inference_mode field
2. `backend/main.py` - Inference mode logic for consensus endpoint
3. `backend/backtesting_engine.py` - Pass cutoff_date, skip training
4. `frontend/src/pages/BacktestPage.jsx` - UI checkbox
5. `examples/crypto_ensemble_forecast.py` - Cutoff_date parameter
6. `strategies/strategy_utils.py` - Cutoff_date parameter
7. `strategies/mean_reversion_strategy.py` - Column name fix

## Testing Performed
- Tested inference mode backtests with proper temporal constraints
- Verified cached models load correctly
- Confirmed no future data leakage in predictions
- Validated Mean Reversion strategy works with both column name cases

## Metrics
- **Prediction Time**: ~7-10 seconds (inference) vs ~3-5 minutes (training mode)
- **Speedup**: ~20-40x faster
- **Model Age**: Can use models days or weeks old for quick testing

## Related PR
Merged in PR #84
